tmt/beakerlib.py

```diff
@@ -302,22 +307,34 @@ def dependencies(
     list of regular rpm package names aggregated from all fetched
     libraries, list of aggregated recommended packages and a list of
     gathered libraries (instances of the Library class).
+
+    Avoid infinite recursion by keeping track of imported library identifiers
+    and not trying to fetch those again.
     """
     # Initialize lists, use set for require & recommend
     processed_require = set()
     processed_recommend = set()
+    imported_lib_ids = imported_lib_ids or []
     gathered_libraries = []
     original_require = original_require or []
     original_recommend = original_recommend or []
 
-    for dependency in original_require + original_recommend:
+    # Cut circular dependencies to avoid infinite recursion
+    def already_fetched(lib: str) -> bool:
```

Comment:
Just naming question: Seems the function is doing the exact opposite to `already_fetched`? Returns `True` if the library is not fetched yet.